synapse/handlers/directory.py

```diff
@@ -325,7 +309,7 @@ def _update_canonical_alias(self, requester, user_id, room_id, room_alias):
         alt_aliases = content.pop("alt_aliases", None)
         # If the aliases are not a list (or not found) do not attempt to modify
         # the list.
-        if isinstance(alt_aliases, list):
+        if isinstance(alt_aliases, (list, tuple)):
```

Comment:
While testing this branch I realized that this code added in #6904 will likely want to handle this as a tuple, not a list. Seemed prudent to accept both.

Reply:
oh yes, because sometimes we `freeze` events, which turns lists into tuples.

for similar situations with dicts we've done `isinstance(d, collections.Mapping)`. I guess the equivalent here would be `isinstance(alt_aliases, collections.Sequence)`. ymmv though.

Reply:
👍 Good call. I made this change.